src/Http/Requests/GetCourseAPIRequest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace EscolaLms\Courses\Http\Requests;
+
+use EscolaLms\Courses\Models\Course;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Facades\Gate;
+
+class GetCourseAPIRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        $user = auth()->user();
+        $course = $this->getCourse();
+        if (is_null($course)) {
+            return true; // controller will fire 404 error
+        }
+        return Gate::check('view', $course);
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [];
+    }
+
+    public function getCourse(): ?Course
+    {
+        return Course::find($this->route('course'));
+    }
+
+    public function userIsUnprivileged(): bool
+    {
+        return empty($this->user()) || $this->user()->cannot('update', $this->getCourse());
```

Comment:
this is `get`/ `view` but you're checking `update`

also course GET `/course/:id` should be available to anybody since this is a preview o course eg https://lms-demo.stage.etd24.pl/courses/6